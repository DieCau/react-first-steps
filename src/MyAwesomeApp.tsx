// Importar para mis estilos las CSSProperties
import type { CSSProperties } from 'react';
// import { test } from 'vitest';

// Debo tener las definiciones de mis variables por fuera del componente
const firstName = 'Diego';
const lastName = 'Caucota';
const favoriteGames = ['Fifa 2025', 'Roblox', 'Minecraft'];
const isActive = false;
const address = {
  zipCode: 'ABS-123',
  country: 'Argentina',
};
const myStyles: CSSProperties = {
  backgroundColor: '#ebb6b6ff',
  borderRadius: isActive ? 10 : 20,
  padding: 10,
  marginTop: 50,
};

// Testear componente
export const MyAwesomeApp = () => {
  return (
    <>
      <h1 data-testid='first-name-title'> {firstName} </h1>
      <h3> {lastName} </h3>
      <p> {favoriteGames.join(', ')} </p>
      <h1>{isActive ? 'Activo' : 'No activo'}</h1>
      <p style={myStyles}>{JSON.stringify(address)}</p>
    </>
  );
};
