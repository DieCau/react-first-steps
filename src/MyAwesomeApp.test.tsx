// Configuraciones generales
// Montar el componente para testearlo
// Instalar Testing Library y usar "render"
// Realizar configuraciones en vite.config.ts

import { describe, expect, test } from 'vitest';

import { render, screen } from '@testing-library/react';
import { MyAwesomeApp } from './MyAwesomeApp';

describe('MyAwesomeApp', () => {
  test('should render firstName and lastName', () => {
    const { container } = render(<MyAwesomeApp />);
    // screen.debug();

    // ! ASSERT
    const h1 = container.querySelector('h1');
    const h3 = container.querySelector('h3');

    // "toContain" es que contenga esa palabra
    expect(h1?.innerHTML).toContain('Diego');
    expect(h3?.innerHTML).toContain('Caucota');
  });

  test('should render firstName and lastName - screen', () => {
    // No tenemos la necesidad de llamar al container
    render(<MyAwesomeApp />);
    // screen.debug();

    //! ASSERT
    // En lugar de buscarlo por el rol
    // const h1 = screen.getByRole('heading', {
    //   level: 1,
    // });

    // Buscar por el testId
    const h1 = screen.getByTestId('first-name-title');

    expect(h1.innerHTML).toContain('Diego');
  });

  // Verificar que el DOM no fue alterado con "toMatchSnapshot"
  //! Recordar: si los cambios son correctos presione "u" y se actualiza
  test('should match snapshot', () => {
    const { container } = render(<MyAwesomeApp />);
    expect(container).toMatchSnapshot();
  });

  test('should match snapshot', () => {
    render(<MyAwesomeApp />);
    expect(screen.getByTestId('div-app')).toMatchSnapshot();
  });
});
