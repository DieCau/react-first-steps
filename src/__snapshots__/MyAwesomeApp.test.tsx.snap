// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MyAwesomeApp > should match snapshot 1`] = `
<div>
  <div
    data-testid="div-app"
  >
    <h1
      data-testid="first-name-title"
    >
       
      Diego
       
    </h1>
    <h3>
       
      Caucota
       
    </h3>
    <p>
      4
    </p>
    <p
      class="mi-clase-favorita"
    >
       
      Fifa 2025, Roblox, Minecraft
       
    </p>
    <h1>
      No activo
    </h1>
    <p
      style="background-color: rgb(235, 182, 182); border-radius: 20px; padding: 10px; margin-top: 50px;"
    >
      {"zipCode":"ABS-123","country":"Argentina"}
    </p>
  </div>
</div>
`;

exports[`MyAwesomeApp > should match snapshot 2`] = `
<div
  data-testid="div-app"
>
  <h1
    data-testid="first-name-title"
  >
     
    Diego
     
  </h1>
  <h3>
     
    Caucota
     
  </h3>
  <p>
    4
  </p>
  <p
    class="mi-clase-favorita"
  >
     
    Fifa 2025, Roblox, Minecraft
     
  </p>
  <h1>
    No activo
  </h1>
  <p
    style="background-color: rgb(235, 182, 182); border-radius: 20px; padding: 10px; margin-top: 50px;"
  >
    {"zipCode":"ABS-123","country":"Argentina"}
  </p>
</div>
`;
